engine/Shopware/Controllers/Backend/Order.php

```diff
@@ -685,17 +701,21 @@ public function batchProcessAction()
             return;
         }
 
-        foreach ($orders as $key => $data) {
-            $orders[$key]['mail'] = null;
-            $orders[$key]['languageSubShop'] = null;
+        foreach ($orders as &$data) {
+            $data['mail'] = null;
+            $data['languageSubShop'] = null;
 
-            if (empty($data) || empty($data['id'])) {
+            if (empty($data['id'])) {
+                $data['success'] = false;
+                $data['errorMessage'] = $namespace->get('no_order_id_passed', 'No valid order id passed.');
```

Comment:
Is this really an error we need to report back to the user? From reading the code, this looks like a sanity check but shouldn't really happen in production?